packages/shared/lib/sdk/sync.ts

```diff
@@ -691,19 +691,26 @@ export class NangoAction {
     }
 
     private async sendLogToPersist(content: string, options: { level: LogLevel; timestamp: number }) {
-        const response = await this.persistApi({
-            method: 'POST',
-            url: `/environment/${this.environmentId}/log`,
-            headers: {
-                Authorization: `Bearer ${this.nango.secretKey}`
-            },
-            data: {
-                activityLogId: this.activityLogId,
-                level: options.level ?? 'info',
-                timestamp: options.timestamp,
-                msg: content
-            }
-        });
+        let response: AxiosResponse;
+        try {
+            response = await this.persistApi({
+                method: 'POST',
+                url: `/environment/${this.environmentId}/log`,
+                headers: {
+                    Authorization: `Bearer ${this.nango.secretKey}`
+                },
+                data: {
+                    activityLogId: this.activityLogId,
+                    level: options.level ?? 'info',
+                    timestamp: options.timestamp,
+                    msg: content
+                }
+            });
+        } catch (err) {
+            logger.error('Failed to log to persist, due to an internal error', err);
+            // TODO: find a way to report that
+            return;
```

Comment:
Though one. Failing silently allows the script to continue when logging fails but there is no way to report to the user since the logging failed. I think it is fine to fail silently in order not to stop the script. Maybe, in addition to the TODO add a comment to explain our reasoning for now

Reply:
added a comment